packages/client/src/components/app/table/Table.svelte

```diff
@@ -55,32 +56,52 @@
     }
   }
 
-  const getFields = (schema, customColumns, showAutoColumns) => {
-    // Check for an invalid column selection
-    let invalid = false
-    customColumns?.forEach(column => {
-      const columnName = typeof column === "string" ? column : column.name
-      if (schema[columnName] == null) {
-        invalid = true
-      }
-    })
-
-    // Use column selection if it exists
-    if (!invalid && customColumns?.length) {
+  const getFields = (
+    schema,
+    customColumns,
+    showAutoColumns,
+    primaryDisplay
+  ) => {
+    if (customColumns?.length) {
       return customColumns
     }
 
     // Otherwise generate columns
     let columns = []
     let autoColumns = []
     Object.entries(schema).forEach(([field, fieldSchema]) => {
+      if (!fieldSchema.visible) {
+        return
+      }
       if (!fieldSchema?.autocolumn) {
```

Comment:
Would this need to default to true when not set to ensure its visible?

Reply:
Nice one - I think I actually meant to check against an explicit `false`, which would mean nullish is considered visible.

Reply:
Pushed that up now - would you mind checking if that now works as expected?

Reply:
Reviewed this again, all default visibility checks for the Table/Tableblock working as expected 👍